src/corehost/cli/bdl_file_entry.cpp

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifdef FEATURE_APPHOST
+
+#include "bdl_processor.h"
+#include "pal.h"
+#include "error_codes.h"
+#include "trace.h"
+#include "utils.h"
+
+bool file_entry_t::is_valid()
+{
+	return data.offset > 0 && data.size > 0 && 
+		   (file_type_t)data.type < file_type_t::END &&
+		   data.path_length > 0 && data.path_length <= PATH_MAX;
+}
+
+file_entry_t* file_entry_t::read(FILE* bundle)
```

Comment:
`bundle` is not a good name here - too overloaded. `bundleFile` at least, ideally something without `bundle` completely (though I don't have a good idea right now).